Feature/fetching api and working with shortening functionality

marcinel123 opened the pull request:
Hi Adam,

In this PR you can find following added features:

- added axios to the project to fetch api
- added functions to handle input and button to shorten long links
- also added functionality so it can work with localStorage
- added new component LinksList to show every new shorten link
- still working on next features like: validating input field and showing error if input empty.

please have a look and let me know
Martin

marcinel123 committed:
added axios and js logic to shorten the links. also links works with localStorage

marcinel123 committed:
links list working with single elements. long links, short links and button visible in every new element

AdamKniec commented on app/src/components/InputElement/InputElement.styles.js:
```diff
@@ -66,3 +66,16 @@ export const StyledInputField = styled.input`
 `;
+
+export const StyledLinksList = styled.div`
+  @media (min-width: 350px) {
```
i thought this breakpoint will be removed ;)

marcinel123 replied:
fixed

AdamKniec commented:
<img width="1504" alt="image" src="https://user-images.githubusercontent.com/20644432/217304446-c16cbe54-5262-4752-ac28-d42933358400.png">
I think that this should be fixed

AdamKniec commented:
Also, you should add validation so that user does not submit the form with no data. You can do it in the next PR if you want ;)

AdamKniec commented on app/src/components/InputElement/InputElement.js:
```diff
@@ -1,21 +1,67 @@
-      <StyledButton>Shorten It!</StyledButton>
-    </StyledForm>
+    <>
+      <StyledForm>
```
Form is not connected with a label

AdamKniec commented on app/src/components/InputElement/InputElement.js:
```diff
@@ -1,21 +1,67 @@
-      />
-      <StyledButton>Shorten It!</StyledButton>
-    </StyledForm>
+    <>
```
why did You use <> here?

marcinel123 replied:
fixed

AdamKniec commented on app/src/components/Statistics/Statistics.js:
```diff
@@ -10,9 +11,10 @@ import {
+export const Statistics = ({ linksList }) => {
   return (
     <StyledWrapper>
+      <LinksList linksList={linksList} />
```
This can be done in a better way. I think You should create the entirely new section wrapping Form and List and render this section under "Content" section. Make the background color the same as in Statistics so that it looks like one section 👍

AdamKniec commented on app/src/App.js:
```diff
@@ -9,17 +9,20 @@ import { Statistics } from "./components/Statistics/Statistics";
         <GlobalStyles />
         <StyledWrapper>
           <Header />
           <Content />
```
this component should not be called Content. It's more something like "ImageIntroSection"

marcinel123 replied:
fixed

AdamKniec commented on app/src/components/LinksList/LinksList.js:
```diff
@@ -0,0 +1,23 @@
+  StyledLinkandBtnWrapper,
+} from "./LinksList.styles";
+
+export const LinksList = ({ linksList }) => {
```
looks nice and clean 👍

marcinel123 replied:
:)

AdamKniec commented on app/src/components/InputElement/InputElement.styles.js:
```diff
@@ -66,3 +66,16 @@ export const StyledInputField = styled.input`
+    width: 80%;
+    background-color: ${(props) => props.theme.colors.cyan};
+    color: ${(props) => props.theme.colors.white};
+    border: 1px solid black;
```
take color from theme object

marcinel123 replied:
fixed.

AdamKniec commented on app/src/components/InputElement/InputElement.js:
```diff
@@ -1,21 +1,67 @@
+    }
+  }, []);
+
+  const handleInput = (e) => {
```
this can be shortened to be in one line

marcinel123 replied:
changed.

marcinel123 committed:
ended linksList section and added validation to input element

marcinel123 committed:
added new urldata component. also fixed form element to be more responsive

marcinel123 commented:
Hi Adam.

I have changed copy button useState logic accordingly to your instructions. Also added different background color if button has been pressed. I fixed from element when user tries to send empty url and error paragraph shows.

Form and LinksList are now placed in the whole new component UrlData.

I believe most issues has been resolved. If I forgot about anything sorry but not intentionally :)

Regards,
Martin

AdamKniec reviewed (approved):
nice job!

marcinel123 committed:
fixed dropdown menu and moved axios to UrlDataSection
